gen.py

```diff
@@ -112,6 +113,25 @@ class EmptyTopLevelPlotFlags(BaseModel):
     __root__: None = None
 
 
+class TopLevelArtifactFlags(BaseModel):
+    type: str = Field(None, description="Type of the artifact")
+    path: str = Field(None, description="Path to the artifact")
```

Comment:
Shouldn't `path` be required? What is an artifact without a path?

Reply:
Good question, but since we're not going to check/use this `path` in DVC (only in Studio), I assume it's up to user whether he wants to add `path` or not. In the case of DVCLive we're going to add it OFC, since it's mandatory for user to pass it into.

Reply:
>  I assume it's up to user whether he wants to add path or not.

But what does an artifact without a `path` mean, in any context?

Reply:
This can be something you produce in your repo or subdir. E.g.:
1. You register a new version of `myart` and push git tag
2. CI starts, builds your artifact (let's say docker image) and pushes it somewhere (let's say to docker registry)
3. Now you can get it from Docker Registry. In repo you could have a `description` and `labels` that make it to the Docker image's description in Docker Registry, or control the CI execution (e.g. label `customerA` cause pushing the image to `customerA/myimage:latest`, etc).

I understand the example is totally made up though, can try to come up with something more decent.

Reply:
@daavoo, as we discussed yesterday - I made DVC use `name` instead of `path` if the latter isn't given when you're reading artifacts (`dvc.repo.Repo(".").artifacts.read()`), but you still may not fill the `path` as a separate attribute here.

Reply:
Made `path` necessary.